### PR TITLE
Add outputs of eio and warning for PerformancePrecisionTradeoffs 

### LINKED ISSUE #7646: PerformancePrecisionTradeoffs should produce eio output to indicate which options are active
Issue overview
--------------
PerformancePrecisionTradeoffs options should be echoed to the eio output, similar to other global options such as ShadowCalculation and HeatBalanceAlgorithm.

PerformancePrecisionTradeoffs should probably also throw one warning if any non-default options are selected.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
Fixes #7646.
The option of PerformancePrecisionTradeoffs is added in the eio output. A warning message is provided when the input is different from default.
A test file is uploaded.

### Diffs
eio diffs are expected for every file.
err diffs are expected for CarrollMRT-RefBldgLargeOfficeNew2004_Chicago.idf and MultiSpeedHeatPump_DirectSolution - new warning.
results diffs are possibly expected for CarrollMRT-RefBldgLargeOfficeNew2004_Chicago.idf (turned off the coil direct solution for this file).

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [x] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/SimulationManager.cc
Comment (mjwitte) [inline src/EnergyPlus/SimulationManager.cc]: @lgu1234 I was expecting the output to look like this, similar to Simulation Control:
```
! <Performance Precision Tradeoffs>, Use Coil Direct Simulation, Zone Radiant Exchange Algorithm
Performance Precision Tradeoffs, Yes, ScriptF
```
Also, we're no longer using gio::write - see elsewhere in the same source file after merging in Develop (see #7746).

And this should really be printed later in the file, I would put it right after SimulationControl.

File: src/EnergyPlus/SimulationManager.cc
Comment (mjwitte) [inline src/EnergyPlus/SimulationManager.cc]: As this object grows, I'm not sure if we want to have specific warnings for each option, or just write a generic warning if one or more tradeoff options are active? Something like 
"PerformancePrecisionTradoffs: One or more performance overrides have been selected, see the eio output for more details." 
@JasonGlazer @xuanluo113 ?

File: src/EnergyPlus/SimulationManager.cc
Comment (lgu1234) [reply src/EnergyPlus/SimulationManager.cc]: @mjwitte I am going to make changes to "print(outputFiles.eio...". Since I am not familiar with Zone Radiant Exchange Algorithm, I will write Use Coil Direct Simulation first and let other people add more options.

**Review by mjwitte — APPROVED** [review]
Tested with idf and ran all unit tests.
Added output and warning message for radiant exchange model.
Added entry to OutputRules doc.

@JasonGlazer will extend further as part of #7761.

File: testfiles/CarrollMRT-RefBldgLargeOfficeNew2004_Chicago.idf
Comment (mjwitte) [inline testfiles/CarrollMRT-RefBldgLargeOfficeNew2004_Chicago.idf:99]: Noticed that the CarrollMRT idf had the coil direct solution active. Turned this off per @nealkruis

**Comment by mjwitte** [comment]
Not waiting for Windows CI to chime in. Merging.

**Pull request merged**
